decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.strip.capitalize
```

Comment:
[OPTIONAL] I did not get the word `strip`. Could you please explain it to me?

Reply:
it used to remove leading and trailing whitespace on strings